packages/app-layout/src/vaadin-app-layout.js

```diff
@@ -440,7 +440,11 @@ class AppLayout extends ElementMixin(ThemableMixin(ControllerMixin(PolymerElemen
 
     this._navbarSizeObserver = new ResizeObserver(() => {
       requestAnimationFrame(() => {
-        this._blockAnimationUntilAfterNextRender();
+        // Ignore resize caused by drawer toggle click
+        // to make drawer open / close transition work
+        if (!this.__drawerToggleClicked) {
+          this._blockAnimationUntilAfterNextRender();
+        }
         this._updateOffsetSize();
```

Comment:
This would have the side-effect of running `_updateOffsetSize` each animation frame. Might not be a big deal, but I think this is why I added the animation blocking in https://github.com/vaadin/web-components/pull/4709.

Also I wonder if we can remove the animation blocking completely if we start working around it in such a way? The `requestAnimationFrame` around it was also only added as a workaround because of it: https://github.com/vaadin/web-components/pull/5575

Reply:
> This would have the side-effect of running _updateOffsetSize each animation frame

Blocking would be still there on navbar size change. Only toggling drawer needs to have animation unblocked.

I agree that current blocking logic is very unclear. Maybe we can get rid of it altogether and instead set `animate` attribute to only animate when drawer opens / closes. I will try if this approach works.

Reply:
> Blocking would be still there on navbar size change. Only toggling drawer needs to have animation unblocked.

Just to be clear, the nav bar size changes when you toggle the drawer if you set `primary-section="drawer"`.

Reply:
Yes, I understand. So are you suggesting to remove `this._blockAnimationUntilAfterNextRender();` call?

Reply:
Changed to use debouncer for `_updateOffsetSize()` and removed blocking animation.